src/components/Input.js

```diff
@@ -0,0 +1,43 @@
+import { useContext, useState } from "react";
+import { TodosListContext } from "../contexts";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+export default function Input() {
+  const { setData } = useContext(TodosListContext);
+  const [inputValue, setInputValue] = useState("");
+  function addNewTodo(title) {
+    setData((data) => [
+      ...data,
+      { id: data[data.length - 1]?.id + 1 || 1, title: title },
+    ]);
+    setInputValue("");
+  }
```

Comment:
You can move this function logic in the context as well, and only expose the function.